tests/integration/plugins/test_npm.py

```diff
@@ -166,3 +183,69 @@ def test_npm_plugin_include_node(new_dir, partitions):
     # try to use bundled Node.js to execute the script
     output = subprocess.check_output([str(node_path), str(binary)], text=True)
     assert output == "hello world\n"
+
+
+@pytest.mark.slow
+def test_npm_plugin_get_file_list(create_fake_package_with_node, new_dir, partitions):
+    parts = create_fake_package_with_node()
+    lifecycle = LifecycleManager(
+        parts,
+        application_name="test_npm_plugin",
+        cache_dir=new_dir,
+        partitions=partitions,
+    )
+    actions = lifecycle.plan(Step.BUILD)
+
+    with lifecycle.action_executor() as ctx:
+        ctx.execute(actions)
+
+    part_name = list(parts["parts"].keys())[0]
+    actual_file_list = lifecycle._executor._handler[part_name]._plugin.get_file_list()
+
+    # This example bundles in node, which brings a ton of other dependencies -
+    # this is perfect for checking all sorts of weird behavior.
+
+    for (pkg_name, _), pkg_files in actual_file_list.items():
```

Comment:
the way this loop is structured is too fragile: if the file list for some reason doesn't have `npm-hello` or `npm` it will still succeed.

Reply:
https://github.com/canonical/craft-parts/pull/931/commits/d385065bf4b95b769d2be393af8bbf4e3ce78e8d